sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/implementation/SparkBridgeImplementationInternal.scala

```diff
@@ -192,6 +195,39 @@ private[cosmos] object SparkBridgeImplementationInternal extends BasicLoggingTra
     rangeToNormalizedRange(feedRange.getEffectiveRange(pkDefinition))
   }
 
+  private[cosmos] def partitionKeyValuesToNormalizedRange
+  (
+      partitionKeyValue: Object,
+      partitionKeyDefinitionJson: String
+  ): NormalizedRange = {
+
+      val partitionKey = new PartitionKeyBuilder()
+      val objectMapper = new ObjectMapper()
+      val json = partitionKeyValue.toString
+      var isMultiplePartitionKeyPaths = false
+      try {
+          val partitionKeyValues = objectMapper.readValue(json, classOf[Array[String]])
+          for (value <- partitionKeyValues) {
+              partitionKey.add(value.trim)
+          }
+          partitionKey.build()
+          isMultiplePartitionKeyPaths = true
+      } catch {
+          case e: Exception =>
+              logInfo("Invalid partition key paths: " + json, e)
+      }
+      if (!isMultiplePartitionKeyPaths) {
```

Comment:
Figuring out whether this is a single or multiple partition key by trying and catching exception is super inefficient - same as above here - please tighten the API contract

Reply:
Oh, I don't know why I have this check. I think this is from when I tried to make changes to the existing UDF, instead of creating a new one for multi-partition keys. Will remove.

Reply:
Done